diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt

```diff
@@ -0,0 +1,79 @@
+package org.cqfn.diktat.ruleset.utils
+
+import com.pinterest.ktlint.core.ast.ElementType.BLOCK
+import org.jetbrains.kotlin.cli.common.CLIConfigurationKeys
+import org.jetbrains.kotlin.cli.common.environment.setIdeaIoUseFallback
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import org.jetbrains.kotlin.cli.jvm.compiler.EnvironmentConfigFiles
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.mock.MockProject
+import org.jetbrains.kotlin.com.intellij.openapi.Disposable
+import org.jetbrains.kotlin.com.intellij.openapi.util.UserDataHolderBase
+import org.jetbrains.kotlin.com.intellij.pom.PomModel
+import org.jetbrains.kotlin.com.intellij.pom.PomModelAspect
+import org.jetbrains.kotlin.com.intellij.pom.PomTransaction
+import org.jetbrains.kotlin.com.intellij.pom.impl.PomTransactionBase
+import org.jetbrains.kotlin.com.intellij.pom.tree.TreeAspect
+import org.jetbrains.kotlin.com.intellij.psi.TokenType.ERROR_ELEMENT
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import sun.reflect.ReflectionFactory
+
+class KotlinParser {
+
+    /**
+     * Set idea.io.use.nio2 in system property to true
+     * fixme Maybe should check OS to don't change system property
+     */
+    init {
+        setIdeaIoUseFallback()
+    }
+
+    companion object{
+        private const val ONE_CHILD = 1
+    }
+
+    fun createNode(text: String, isPackage: Boolean = false): ASTNode {
```

Comment:
Please add documentation to this class, including some not obvious things we discussed, e.g. `createFile` is called only for `isPackage`.